src/utils/worker/createResponseListener.ts

```diff
@@ -12,14 +12,26 @@ export function createResponseListener(context: SetupWorkerInternalContext) {
       ServiceWorkerIncomingEventsMap['RESPONSE']
     >,
   ) => {
-    const { payload } = message
-    const response = new Response(payload.body, payload)
+    const { payload: responseJson } = message
+
+    /**
+     * CORS requests with `mode: "no-cors"` result in "opaque" responses.
+     * That kind of responses cannot be manipulated in JavaScript due
+     * to the security considerations.
+     * @see https://fetch.spec.whatwg.org/#concept-filtered-response-opaque
+     * @see https://github.com/mswjs/msw/issues/529
+     */
+    if (responseJson.type.includes('opaque')) {
```

Comment:
I could suggest adding an extra check here. Because the `type` attribute is new, some users could not have the `service worker` updated and will receive a lot of errors. 
```suggestion
    if ("type" in responseJson && responseJson.type.includes('opaque')) {
```

Reply:
Yes, this is a valid concern, but as I've mentioned [in the comment above](https://github.com/mswjs/msw/pull/564#issuecomment-766327879), this is going to be released as a breaking change. So there's no need to provide a backward-compatibility hatch.

Reply:
okok so it's ok for me 🥳